infrastructure/tooling/src/importer/importer.js

```diff
@@ -19,67 +20,134 @@ const importer = async options => {
     }
   }
 
-  for (let tableName of tables) {
-    tasks.push({
-      title: `Import Table ${tableName}`,
-      locks: ['concurrency'],
-      requires: [],
-      provides: [tableName],
-      run: async (requirements, utils) => {
-        const start = new Date();
-
-        const pgTable = postgresTableName(tableName);
-
-        await db._withClient('admin', async client => {
-          await client.query(`truncate ${pgTable}`);
-        });
+  async function* readAzureTableIterator(args) {
+    yield await readAzureTableInChunks(args);
+  }
 
-        async function* readAzureTableIterator(args) {
-          yield await readAzureTableInChunks(args);
+  async function importTable(tableName, tableParameters = {}, rowsProcessed = 0, utils) {
+    for await (let result of buffer(
+      readAzureTableIterator({
+        azureCreds: credentials.azure,
+        tableName,
+        utils,
+        tableParams: tableParameters,
+        rowsProcessed,
+      }),
+      1000,
+    )) {
+      if (result) {
+        const { entities, tableParams, count } = result;
+
+        await writeToPostgres(tableName, entities, db, ALLOWED_TABLES);
+
+        if (tableParams.nextPartitionKey && tableParams.nextRowKey) {
+          return await importTable(tableName, tableParams, count, utils);
         }
 
-        async function importTable(tableParameters = {}, rowsProcessed = 0) {
-          for await (let result of buffer(
-            readAzureTableIterator({
-              azureCreds: credentials.azure,
-              tableName,
-              utils,
-              tableParams: tableParameters,
-              rowsProcessed,
-            }),
-            1000,
-          )) {
-            if (result) {
-              const { entities, tableParams, count } = result;
-
-              await writeToPostgres(tableName, entities, db, ALLOWED_TABLES);
-
-              if (tableParams.nextPartitionKey && tableParams.nextRowKey) {
-                return await importTable(tableParams, count);
-              }
-
-              return count;
-            }
-
-            return 0;
-          }
-        }
+        return count;
+      }
 
-        const rowsImported = await importTable();
+      return 0;
+    }
+  }
 
-        return {
-          [tableName]: {
-            elapsedTime: new Date() - start,
-            rowsImported,
+  let largeTableNames = [];
+  for (let tableName of tables) {
+    if (LARGE_TABLES.includes(tableName)) {
+      const pgTable = postgresTableName(tableName);
+
+      await db._withClient('admin', async client => {
+        await client.query(`truncate ${pgTable}`);
+      });
+      const may2020 = new Date(2020, 4, 1);
+      const july2020 = new Date(2020, 6, 1);
+      const september2020 = new Date(2020, 8, 1);
+      const november2020 = new Date(2020, 10, 1);
+      const january2021 = new Date(2020, 12, 1);
```

Comment:
I am assuming that QueueTasks and QueueArtifacts entries expire after 1 year. Let me know if this is incorrect.

Reply:
I think Firefox CI sets expirations of at most at 1 year, yes.  But this isn't a general characteristic of Taskcluster.

Reply:
I think that's fine then. As long as most things expire in 1 year, this should speed up the import.